util/pace/util/grid/generation.py

```diff
@@ -2351,6 +2605,129 @@ def _init_cell_trigonometry(self):
+            [util.X_DIM, util.Y_DIM],
+            "",
+            dtype=np.float64,
+            allow_mismatch_float_precision=True,
+        )
+        sina_u_64 = self.quantity_factory.zeros(
+            [util.X_INTERFACE_DIM, util.Y_DIM],
+            "",
+            dtype=np.float64,
+            allow_mismatch_float_precision=True,
+        )
+        sina_v_64 = self.quantity_factory.zeros(
+            [util.X_DIM, util.Y_INTERFACE_DIM],
+            "",
+            dtype=np.float64,
+            allow_mismatch_float_precision=True,
+        )
+        rsin_u_64 = self.quantity_factory.zeros(
+            [util.X_INTERFACE_DIM, util.Y_DIM],
+            "",
+            dtype=np.float64,
+            allow_mismatch_float_precision=True,
+        )
+        rsin_v_64 = self.quantity_factory.zeros(
+            [util.X_DIM, util.Y_INTERFACE_DIM],
+            "",
+            dtype=np.float64,
+            allow_mismatch_float_precision=True,
+        )
+        rsina_64 = self.quantity_factory.zeros(
+            [util.X_INTERFACE_DIM, util.Y_INTERFACE_DIM],
+            "",
+            dtype=np.float64,
+            allow_mismatch_float_precision=True,
+        )
+        rsin2_64 = self.quantity_factory.zeros(
+            [util.X_DIM, util.Y_DIM],
+            "",
+            dtype=np.float64,
+            allow_mismatch_float_precision=True,
+        )
+        cosa_64 = self.quantity_factory.zeros(
+            [util.X_INTERFACE_DIM, util.Y_INTERFACE_DIM],
+            "",
+            dtype=np.float64,
+            allow_mismatch_float_precision=True,
+        )
+        sina_64 = self.quantity_factory.zeros(
+            [util.X_INTERFACE_DIM, util.Y_INTERFACE_DIM],
+            "",
+            dtype=np.float64,
+            allow_mismatch_float_precision=True,
+        )
+
+        cosa_u_64.data[:, :] = 0.0
+        cosa_v_64.data[:, :] = 0.0
+        cosa_s_64.data[:, :] = 0.0
+        sina_u_64.data[:, :] = 1.0
+        sina_v_64.data[:, :] = 1.0
+        rsin_u_64.data[:, :] = 1.0
+        rsin_v_64.data[:, :] = 1.0
+        rsina_64.data[:, :] = 1.0
+        rsin2_64.data[:, :] = 1.0
+        cosa_64.data[:, :] = 0.0
+        sina_64.data[:, :] = 1.0
+
+        for i in range(1, 10):
+            sin_sg = self.quantity_factory.zeros(
+                [util.X_DIM, util.Y_DIM],
+                "",
+                dtype=np.float64,
+                allow_mismatch_float_precision=True,
+            )
+            sin_sg.data[:, :] = 1.0
+            setattr(
+                self,
+                f"_sin_sg{i}",
+                quantity_cast_to_model_float(self.quantity_factory, sin_sg),
+            )
+            if i == 5:
+                self._sin_sg5_64 = sin_sg
+            cos_sg = self.quantity_factory.zeros(
+                [util.X_DIM, util.Y_DIM],
+                "",
+                dtype=np.float64,
+                allow_mismatch_float_precision=True,
+            )
+            cos_sg.data[:, :] = 0.0
```

Comment:
Similarly, if we initialize these as zeroes and ones I think we can skip most of the assignments here:
```suggestion
    def _init_cell_trigonometry_cartesian(self):

        cosa_u_64 = self.quantity_factory.zeros(
            [util.X_INTERFACE_DIM, util.Y_DIM],
            "",
            dtype=np.float64,
            allow_mismatch_float_precision=True,
        )
        cosa_v_64 = self.quantity_factory.zeros(
            [util.X_DIM, util.Y_INTERFACE_DIM],
            "",
            dtype=np.float64,
            allow_mismatch_float_precision=True,
        )
        cosa_s_64 = self.quantity_factory.zeros(
            [util.X_DIM, util.Y_DIM],
            "",
            dtype=np.float64,
            allow_mismatch_float_precision=True,
        )
        sina_u_64 = self.quantity_factory.ones(
            [util.X_INTERFACE_DIM, util.Y_DIM],
            "",
            dtype=np.float64,
            allow_mismatch_float_precision=True,
        )
        sina_v_64 = self.quantity_factory.ones(
            [util.X_DIM, util.Y_INTERFACE_DIM],
            "",
            dtype=np.float64,
            allow_mismatch_float_precision=True,
        )
        rsin_u_64 = self.quantity_factory.ones(
            [util.X_INTERFACE_DIM, util.Y_DIM],
            "",
            dtype=np.float64,
            allow_mismatch_float_precision=True,
        )
        rsin_v_64 = self.quantity_factory.ones(
            [util.X_DIM, util.Y_INTERFACE_DIM],
            "",
            dtype=np.float64,
            allow_mismatch_float_precision=True,
        )
        rsina_64 = self.quantity_factory.ones(
            [util.X_INTERFACE_DIM, util.Y_INTERFACE_DIM],
            "",
            dtype=np.float64,
            allow_mismatch_float_precision=True,
        )
        rsin2_64 = self.quantity_factory.ones(
            [util.X_DIM, util.Y_DIM],
            "",
            dtype=np.float64,
            allow_mismatch_float_precision=True,
        )
        cosa_64 = self.quantity_factory.zeros(
            [util.X_INTERFACE_DIM, util.Y_INTERFACE_DIM],
            "",
            dtype=np.float64,
            allow_mismatch_float_precision=True,
        )
        sina_64 = self.quantity_factory.ones(
            [util.X_INTERFACE_DIM, util.Y_INTERFACE_DIM],
            "",
            dtype=np.float64,
            allow_mismatch_float_precision=True,
        )

        for i in range(1, 10):
            sin_sg = self.quantity_factory.ones(
                [util.X_DIM, util.Y_DIM],
                "",
                dtype=np.float64,
                allow_mismatch_float_precision=True,
            )
            setattr(
                self,
                f"_sin_sg{i}",
                quantity_cast_to_model_float(self.quantity_factory, sin_sg),
            )
            if i == 5:
                self._sin_sg5_64 = sin_sg
            cos_sg = self.quantity_factory.zeros(
                [util.X_DIM, util.Y_DIM],
                "",
                dtype=np.float64,
                allow_mismatch_float_precision=True,
            )
```